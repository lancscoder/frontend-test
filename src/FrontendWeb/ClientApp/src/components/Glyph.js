import React from 'react';

const Glyph = ({name}) => (
  <span className={name} style={{'paddingLeft': '5px', 'paddingRight': '5px'}} aria-hidden="true"></span>
);

export default Glyph;