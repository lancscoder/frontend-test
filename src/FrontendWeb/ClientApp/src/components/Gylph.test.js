import React from 'react';
import renderer from 'react-test-renderer';
import Glyph from './Glyph';

describe('Glyph ::', () => {
    it('should render with correct class', () => {
        const component = renderer.create(<Glyph name="test" />);;

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        expect(tree.props.className).toBe('test');
    });
});