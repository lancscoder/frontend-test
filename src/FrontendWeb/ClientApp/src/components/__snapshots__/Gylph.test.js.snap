// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Glyph :: should render with correct class 1`] = `
<span
  aria-hidden="true"
  className="test"
  style={
    Object {
      "paddingLeft": "5px",
      "paddingRight": "5px",
    }
  }
/>
`;
